sledgehammer/start-up.sh

```diff
@@ -107,28 +114,43 @@ else
     fi
 fi
 
+IPv6=0
 if ! [[ $(ip -4 -o addr show dev $BOOTDEV) =~ $ip_re ]]; then
-    echo "We did not get an address on $BOOTDEV"
-    echo "Things will end badly."
+    if [[ $(ip -6 -o addr show dev $BOOTDEV) =~ $ip6_re ]]; then
+        IPv6=1
+    else
```

Comment:
This feel wrong, somehow. I don't like the nesting either. I wonder if bash has the concept of !~ like Perl does.

But I also can't think of any better way to implement. :-(

Reply:
yeah, I guess we cloud not nest. But we do want to track the ip_version and we get that from here, so could use some else ifs. But it'll end up logically the same